File: frontend/src/types/climate.ts
export interface MarketIndex {
  symbol: string;
  name: string;
  current_price: number;
  change_percent: number;
  trend: "up" | "down" | "neutral";
  currency: string;
}

export interface SignalDistribution {
  total_stocks: number;
  buy: number;
  sell: number;
  watch: number;
  hold: number;
  avoid: number;
}

export interface InternalSignals {
  signal_distribution: SignalDistribution;
  avg_change_percent: number;
  high_volume_count: number;
  overall_sentiment: "bullish" | "bearish" | "neutral";
}

export interface CalendarEvent {
  id: string;
  date: string;
  title: string;
  category: string;
  source: string;
  impact: "high" | "medium" | "low";
  days_until: number;
  is_active: boolean;
}

export interface ClimateOverview {
  indices: MarketIndex[];
  signals: InternalSignals;
  events: CalendarEvent[];
  timestamp: string;
}

File: backend/schemas/climate.py
from pydantic import BaseModel
from typing import Optional, List


class MarketIndex(BaseModel):
    symbol: str
    name: str
    current_price: float = 0.0
    change_percent: float = 0.0
    trend: str = "neutral"  # "up", "down", "neutral"
    currency: str = "SEK"


class SignalDistribution(BaseModel):
    total_stocks: int = 0
    buy: int = 0
    sell: int = 0
    watch: int = 0
    hold: int = 0
    avoid: int = 0


class InternalSignals(BaseModel):
    signal_distribution: SignalDistribution = SignalDistribution()
    avg_change_percent: float = 0.0
    high_volume_count: int = 0
    overall_sentiment: str = "neutral"  # "bullish", "bearish", "neutral"


class CalendarEvent(BaseModel):
    id: str
    date: str
    title: str
    category: str = ""
    source: str = ""
    impact: str = "medium"  # "high", "medium", "low"
    days_until: int = 0
    is_active: bool = False


class ClimateOverview(BaseModel):
    indices: List[MarketIndex] = []
    signals: InternalSignals = InternalSignals()
    events: List[CalendarEvent] = []
    timestamp: str = ""
